src/model/Content.ts

```diff
@@ -0,0 +1,24 @@
+export interface InlineContent {
+  text: string
+}
+
+export interface LineContent {
+  type: 'line'
+  /** line position index starts from 0 */
+  line: number
+  /** actual visible contents */
+  children?: InlineContent[]
+}
+
+/**
+ * @note
+ * Every `lineBegin` and `lineEnd` of `NestedContent` must be unique
```

Comment:
Note that this might be same within same `NestedContent`. It differs when `NestedContent` is different.